src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs

```diff
@@ -138,31 +138,39 @@ private IEnumerable<SyntaxNode> GetMembersToRemove(SyntaxNode root)
 
                 // get potential namespace, types and members to remove.
                 var removableCandidates = root
-                    .DescendantNodes(n => DescendIntoChildren(n, spine.Contains(n)))
-                    .Where(n => FilterToTopLevelMembers(n, State.TypeNode));
+                    .DescendantNodes(n => spine.Contains(n))
```

Comment:
Yes that explanation is perfect. I had the helper to improve performance, to not descend into every syntax, but that was probably before I introduced `spine` which has only the members we want to descend into.  This is correct.